libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -0,0 +1,366 @@
+{
+   graphene::chain::database& db = database();
+   const auto &stats_obj = account_id(db).statistics(db);
+
+   // add new entry
+   const auto &ath = db.create<account_transaction_history_object>([&](account_transaction_history_object &obj) {
+      obj.operation_id = op_id;
+      obj.account = account_id;
+      obj.sequence = stats_obj.total_ops + 1;
+      obj.next = stats_obj.most_recent_op;
+   });
+
+   // keep stats growing as no op will be removed
+   db.modify(stats_obj, [&](account_statistics_object &obj) {
+      obj.most_recent_op = ath.id;
+      obj.total_ops = ath.sequence;
+   });
+
+   // curl buffers to read
+   std::string readBuffer;
+   std::string readBuffer_logs;
+
+   // ath data
+   std::string account_transaction = fc::json::to_string(ath.to_variant());
+
+   // operation history data
+   std::string trx_in_block = fc::json::to_string(op_id(db).trx_in_block);
+   std::string op_in_trx = fc::json::to_string(op_id(db).op_in_trx);
+
+   std::string operation_result = fc::json::to_string(op_id(db).result);
+   boost::replace_all(operation_result, "\"", "'");
+
+   std::string virtual_op = fc::json::to_string(op_id(db).virtual_op);
+
+   std::string op = fc::json::to_string(op_id(db).op);
+   // escaping some special characters inside op
+   boost::replace_all(op, "'", ""); // '
+   boost::replace_all(op, "\\'", ""); // \'
+   boost::replace_all(op, "\\\"", ""); // \"
+   boost::replace_all(op, "\"", "'"); // replace all " for '
+
+
+
+   std::string op_type = "";
+   if (!op_id(db).id.is_null())
+      op_type = fc::json::to_string(op_id(db).op.which());
+
+   std::string operation_history = "{\"trx_in_block\":" + trx_in_block + ",\"op_in_trx\":" + op_in_trx +
+                                   ",\"operation_results\":\"" + operation_result + "\",\"virtual_op\":" + virtual_op +
+                                   ",\"op\":\"" + op + "\"}";
+
+   // visitor data
+   std::string visitor_data = "";
+   if(_elasticsearch_visitor) {
+      operation_visitor o_v;
+      op_id(db).op.visit(o_v);
+
+      std::string fee_asset = fc::json::to_string(o_v.fee_asset);
+      std::string fee_amount = fc::json::to_string(o_v.fee_amount);
+      boost::replace_all(fee_amount, "\"", "");
+
+      std::string fee_data = "{\"fee_asset\":" + fee_asset + ", \"fee_amount\":\"" + fee_amount + "\"}";
+
+      std::string transfer_asset = fc::json::to_string(o_v.transfer_asset_id);
+      std::string transfer_amount = fc::json::to_string(o_v.transfer_amount);
+      boost::replace_all(transfer_amount, "\"", "");
+
+      std::string transfer_data = "{\"transfer_asset_id\":" + transfer_asset + ", \"transfer_amount\":\"" + transfer_amount + "\"}";
+      visitor_data = ",\"fee_data\": " + fee_data + ", \"transfer_data\": " + transfer_data;
+   }
+
+   // block data
+   auto block = db.fetch_block_by_number(op_id(db).block_num);
```

Comment:
This is an extremely expensive call. Note that you're calling add_elasticsearch only from update_account_histories, and that method already has the signed block as a parameter. You only need to pass it down to add_elasticsearch.

Reply:
nice one, added at https://github.com/bitshares/bitshares-core/pull/444/commits/9ed90c30845d907ad5cb1a632f8492963b2dfa30